### PR TITLE
feat(dgraph): enabling TLS config in http zero (#6691)

### PR DESCRIPTION
**Pull request by aman-bansal**
enabling TLS encryption in zero

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6867)

 

Docs Preview: [Dgraph Preview](https://dgraph-0b3112bec2-107693.surge.sh)

### CODE REVIEW

File: tlstest/zero_https/all_routes_tls/all_routes_tls_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		}
+
+		body := readResponseBody(t, do)
+		if test.response != string(body) {
+			t.Fatalf("response is not same. Got: %s Expected: %s", string(body), test.response)
+		}
+	}
+}
+
+var testCasesHttps = []testCase{
```
[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/all_routes_tls/all_routes_tls_test.go:61]: @codeLingo ignore

File: tlstest/zero_https/all_routes_tls/all_routes_tls_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	url        string
+	statusCode int
+	response   string
+}
+
+var testCasesHttp = []testCase{
```
[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/all_routes_tls/all_routes_tls_test.go:22]: @codelingo ignore

File: tlstest/zero_https/no_tls/no_tls_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	url        string
+	statusCode int
+	response   string
+}
+
+var testCasesHttp = []testCase{
```
[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/no_tls/no_tls_test.go:19]: @codelingo ignore

File: tlstest/zero_https/all_routes_tls/all_routes_tls_test.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+		response:   "\"id\":\"1\",\"addr\":\"zero1:5180\",\"leader\":true",
+		statusCode: 200,
+	},
+}
+
+func TestZeroWithAllRoutesTLSWithTLSClient(t *testing.T) {
+	pool, err := generateCertPool("../../tls/ca.crt", true)
+	require.NoError(t, err)
+
+	tlsCfg := &tls.Config{RootCAs: pool, ServerName: "localhost", InsecureSkipVerify: true}
```
[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/all_routes_tls/all_routes_tls_test.go:78]: @codelingo ignore

[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/all_routes_tls/all_routes_tls_test.go:78]: this is test file

File: tlstest/zero_https/all_routes_tls/all_routes_tls_test.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+func readResponseBody(t *testing.T, do *http.Response) []byte {
+	defer func() { _ = do.Body.Close() }()
+	body, err := ioutil.ReadAll(do.Body)
+	require.NoError(t, err)
+	return body
+}
+
+func generateCertPool(certPath string, useSystemCA bool) (*x509.CertPool, error) {
```
[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/all_routes_tls/all_routes_tls_test.go:112]: @codelingo ignore

[the remark this answers is not in the document]
Comment (aman-bansal) [reply tlstest/zero_https/all_routes_tls/all_routes_tls_test.go:112]: this is test file

File: dgraph/cmd/zero/http.go [hunk 6]
```diff
@@ -239,23 +244,71 @@ func (st *state) pingResponse(w http.ResponseWriter, r *http.Request) {
...
+		IdleTimeout:  2 * time.Minute,
+	}
+
+	err := srv.Serve(l)
+	glog.Errorf("Stopped taking more http(s) requests. Err: %v", err)
+	ctx, cancel := context.WithTimeout(context.Background(), 630*time.Second)
+	defer cancel()
+	err = srv.Shutdown(ctx)
+	glog.Infoln("All http(s) requests finished.")
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (aman-bansal) [reply dgraph/cmd/zero/http.go:302]: @codelingo ignore

**Review by pawanrawal — APPROVED** [review]

**Pull request merged**